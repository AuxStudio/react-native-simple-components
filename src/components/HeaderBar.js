import React from "react";
import { View, Text, StatusBar, StyleSheet, Platform } from "react-native";
import MaterialIcon from "react-native-vector-icons/MaterialIcons";
import { AnimateOpacity } from "react-native-simple-animators";

import styleConstants from "../assets/styleConstants";

import Touchable from "./Touchable";
import DeleteButton from "./DeleteButton";

const iosStyles = Platform.OS === "ios" && {
    paddingTop: 20,
    height: 76,
};

const styles = StyleSheet.create({
    wrapper: {
        alignSelf: "stretch",
    },
    container: {
        height: 56,
        flexDirection: "row",
        justifyContent: "space-between",
        alignItems: "stretch",
        paddingHorizontal: 16,
        backgroundColor: styleConstants.primary,
    },
    leftIconContainer: {
        flex: 1,
        justifyContent: "center",
        alignItems: "flex-start",
    },
    leftIcon: {
        fontSize: styleConstants.iconFont,
        color: styleConstants.white,
    },
    textContainer: {
        flex: 3, // make the text container 3 times as big as the icon containers
        justifyContent: "center",
        alignItems: "center",
    },
    text: {
        fontSize: styleConstants.regularFont,
        color: styleConstants.white,
    },
    rightIconContainer: {
        flex: 1,
        justifyContent: "center",
        alignItems: "flex-end",
    },
    rightIcon: {
        fontSize: styleConstants.iconFont,
        color: styleConstants.white,
    },
});

export default (HeaderBar = props => {
    /* 
        PROPTYPES
    
        backgroundColor: PropTypes.string,
        textColor: PropTypes.string,
        headerShadow: PropTypes.bool, // flag indicating whether or not to show headerShadow
        statusBarStyle: PropTypes.string, // dark-content or light-content
        statusBarColor: PropTypes.string,

        backButton: PropTypes.bool,
        closeButton: PropTypes.bool,
        leftComponent: PropTypes.func,
        leftIconName: PropTypes.string,
        handleLeftIconPress: PropTypes.func,

        textComponent: PropTypes.func,
        text: PropTypes.string,
        textLeft: PropTypes.bool, // aligns text left
        textRight: PropTypes.bool, // aligns text right
        handleTextPress: PropTypes.func,
        // textStyle: PropTypes.node,

        addButton: PropTypes.bool,
        continueButton: PropTypes.bool,
        rightComponent: PropTypes.func,
        rightIconName: PropTypes.string,
        handleRightIconPress: PropTypes.func,
    */

    const backgroundColorStyles = props.backgroundColor && {
        backgroundColor: props.backgroundColor,
    };

    const textColorStyles = props.textColor && {
        color: props.textColor,
    };

    const headerShadowStyles =
        props.headerShadow && styleConstants.regularShadow;

    const leftIcon = props.leftComponent ? (
        <View style={styles.leftIconContainer}>{props.leftComponent()}</View>
    ) : props.leftIconName ? (
        <Touchable
            style={
                props.textLeft
                    ? { justifyContent: "center" }
                    : styles.leftIconContainer
            }
            onPress={props.handleLeftIconPress}>
            <MaterialIcon
                name={props.leftIconName}
                style={[styles.leftIcon, textColorStyles]}
            />
        </Touchable>
    ) : props.backButton ? (
        <Touchable
            style={styles.leftIconContainer}
            onPress={props.handleLeftIconPress}>
            <MaterialIcon
                name="chevron-left"
                style={[styles.leftIcon, textColorStyles]}
            />
        </Touchable>
    ) : props.closeButton ? (
        <Touchable
            style={styles.leftIconContainer}
            onPress={props.handleLeftIconPress}>
            <MaterialIcon
                name="close"
                style={[styles.leftIcon, textColorStyles]}
            />
        </Touchable>
    ) : props.textLeft ? null : (
        <View style={styles.leftIconContainer} />
    );

    const textLeftStyles = props.textLeft
        ? {
              alignItems: "flex-start",
              paddingLeft: 8,
          }
        : null;

    const textRightStyles = props.textRight ? { alignItems: "flex-end" } : null;

    const text = props.textComponent ? (
        props.textComponent()
    ) : props.handleTextPress ? (
        <Touchable
            style={[styles.textContainer, textLeftStyles, textRightStyles]}
            onPress={props.handleTextPress}>
            <Text
                style={[
                    styles.text,
                    styleConstants.primaryFont,
                    textColorStyles,
                    props.textStyle,
                ]}>
                {props.text}
            </Text>
        </Touchable>
    ) : !props.text ? null : (
        <View style={[styles.textContainer, textLeftStyles, textRightStyles]}>
            <Text
                style={[
                    styles.text,
                    styleConstants.primaryFont,
                    textColorStyles,
                    props.textStyle,
                ]}>
                {props.text}
            </Text>
        </View>
    );

    const rightIcon = props.rightComponent ? (
        <View style={styles.rightIconContainer}>{props.rightComponent()}</View>
    ) : props.rightIconName ? (
        <Touchable
            style={styles.rightIconContainer}
            onPress={props.handleRightIconPress}>
            <MaterialIcon
                name={props.rightIconName}
                style={[styles.rightIcon, textColorStyles]}
            />
        </Touchable>
    ) : props.addButton ? (
        <Touchable
            style={styles.rightIconContainer}
            onPress={props.handleRightIconPress}>
            <MaterialIcon
                name="add"
                style={[styles.rightIcon, textColorStyles]}
            />
        </Touchable>
    ) : props.continueButton ? (
        <AnimateOpacity
            initialValue={0}
            finalValue={1}
            shouldAnimateIn
            style={styles.rightIconContainer}>
            <Touchable onPress={props.handleRightIconPress}>
                <MaterialIcon
                    name="check"
                    style={[styles.rightIcon, textColorStyles]}
                />
            </Touchable>
        </AnimateOpacity>
    ) : props.textRight ? null : (
        <View style={styles.rightIconContainer} />
    );

    return (
        <View style={styles.wrapper}>
            <StatusBar
                backgroundColor={
                    props.statusBarColor
                        ? props.statusBarColor
                        : styleConstants.darkTransPrimary
                }
                barStyle={
                    props.statusBarStyle
                        ? props.statusBarStyle
                        : "light-content"
                }
            />
            <View
                style={[
                    styles.container,
                    backgroundColorStyles,
                    headerShadowStyles,
                    iosStyles,
                ]}>
                {leftIcon}
                {text}
                {rightIcon}
            </View>
        </View>
    );
});
