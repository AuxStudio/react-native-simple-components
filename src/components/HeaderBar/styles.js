import { StyleSheet } from 'react-native';

import styleConstants from '../../styleConstants';

const styles = StyleSheet.create({
  wrapper: {
    alignSelf: 'stretch',
  },
  container: {
    height: 56,
    flexDirection: 'row',
    justifyContent: 'space-between',
    alignItems: 'stretch',
    paddingHorizontal: styleConstants.dimensions.padding.large,
    backgroundColor: styleConstants.colors.dividerColor,
  },
  leftIconContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'flex-start',
  },
  leftIcon: {
    fontSize: styleConstants.fonts.sizes.icon,
    color: styleConstants.colors.primaryText,
  },
  textContainer: {
    flex: 5,
    justifyContent: 'center',
    alignItems: 'center',
  },
  text: {
    fontSize: styleConstants.fonts.sizes.regular,
    color: styleConstants.colors.primaryText,
    textAlign: 'center',
  },
  rightIconContainer: {
    flex: 1,
    justifyContent: 'center',
    alignItems: 'flex-end',
  },
  rightIcon: {
    fontSize: styleConstants.fonts.sizes.icon,
    color: styleConstants.colors.primaryText,
  },
});

export default styles;
