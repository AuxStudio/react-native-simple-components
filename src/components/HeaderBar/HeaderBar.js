import React from 'react';
import { View, Text, StatusBar } from 'react-native';

import styles from './styles';
import styleConstants from '../assets/styleConstants';

import MaterialIcon from 'react-native-vector-icons/MaterialIcons';
import { AnimateOpacity } from 'react-native-simple-animators';
import Touchable from './Touchable';

export default (HeaderBar = (props) => {
  /* 
        PROPTYPES
    
        showShadow: PropTypes.bool,
        statusBarStyle: PropTypes.string, // dark-content or light-content
        statusBarColor: PropTypes.string,

        leftComponent: PropTypes.node,
        leftIconName: PropTypes.string,
        handleLeftIconPress: PropTypes.func,

        textComponent: PropTypes.node,
        text: PropTypes.string,
        textLeft: PropTypes.bool, // aligns text left
        textRight: PropTypes.bool, // aligns text right
        handleTextPress: PropTypes.func,

        rightComponent: PropTypes.node,
        rightIconName: PropTypes.string,
        handleRightIconPress: PropTypes.func,

        // leftIconStyle: PropTypes.node,
        // textStyle: PropTypes.node,
        // rightIconStyle: PropTypes.node,
        // style: PropTypes.node,
        // wrapperStyle: PropTypes.node
    */

  const showShadowStyles = props.showShadow && styleConstants.regularShadow;

  const leftIcon = props.leftComponent ? (
    <View style={styles.leftIconContainer}>{props.leftComponent}</View>
  ) : props.leftIconName ? (
    <Touchable
      style={props.textLeft ? { justifyContent: 'center' } : styles.leftIconContainer}
      onPress={props.handleLeftIconPress}
    >
      <MaterialIcon name={props.leftIconName} style={[styles.leftIcon, props.leftIconStyle]} />
    </Touchable>
  ) : props.textLeft ? null : (
    <View style={styles.leftIconContainer} />
  );

  const textLeftStyles = props.textLeft
    ? {
        alignItems: 'flex-start',
        paddingLeft: 8,
      }
    : null;

  const textRightStyles = props.textRight ? { alignItems: 'flex-end' } : null;

  const text = props.textComponent ? (
    props.textComponent
  ) : props.handleTextPress ? (
    <Touchable
      style={[styles.textContainer, textLeftStyles, textRightStyles]}
      onPress={props.handleTextPress}
    >
      <Text style={[styles.text, props.textStyle]} numberOfLines={1}>
        {props.text}
      </Text>
    </Touchable>
  ) : !props.text ? null : (
    <View style={[styles.textContainer, textLeftStyles, textRightStyles]}>
      <Text style={[styles.text, props.textStyle]} numberOfLines={1}>
        {props.text}
      </Text>
    </View>
  );

  const rightIcon = props.rightComponent ? (
    <View style={styles.rightIconContainer}>{props.rightComponent}</View>
  ) : props.rightIconName ? (
    <Touchable style={styles.rightIconContainer} onPress={props.handleRightIconPress}>
      <MaterialIcon name={props.rightIconName} style={[styles.rightIcon, props.rightIconStyle]} />
    </Touchable>
  ) : props.textRight ? null : (
    <View style={styles.rightIconContainer} />
  );

  return (
    <View style={[styles.wrapper, props.wrapperStyle]}>
      <StatusBar
        backgroundColor={
          props.statusBarColor ? props.statusBarColor : styleConstants.darkTransPrimary
        }
        barStyle={props.statusBarStyle ? props.statusBarStyle : 'light-content'}
      />
      <View style={[styles.container, showShadowStyles, iosStyles, props.style]}>
        {leftIcon}
        {text}
        {rightIcon}
      </View>
    </View>
  );
});
