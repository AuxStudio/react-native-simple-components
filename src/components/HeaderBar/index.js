import React from 'react';
import PropTypes from 'prop-types';
import { View, Text, ViewPropTypes, Platform } from 'react-native';
import MaterialIcon from 'react-native-vector-icons/MaterialIcons';

import styles from './styles';
import styleConstants from '../../styleConstants';
import Touchable from '../Touchable';
import StatusBar from '../StatusBar';

const propTypes = {
  showShadow: PropTypes.bool,
  statusBarStyle: PropTypes.string, // dark-content or light-content
  statusBarColor: PropTypes.string,

  leftComponent: PropTypes.node,
  leftIconName: PropTypes.string,
  handleLeftIconPress: PropTypes.func,

  textComponent: PropTypes.node,
  text: PropTypes.string,
  textLeft: PropTypes.bool, // aligns text left
  textRight: PropTypes.bool, // aligns text right
  handleTextPress: PropTypes.func,

  rightComponent: PropTypes.node,
  rightIconName: PropTypes.string,
  handleRightIconPress: PropTypes.func,

  androidRipple: PropTypes.bool,
  androidRippleColor: PropTypes.string,
  androidRippleBorderless: PropTypes.bool,

  leftIconStyle: Text.propTypes.style,
  textStyle: Text.propTypes.style,
  rightIconStyle: Text.propTypes.style,
  style: ViewPropTypes.style,
  wrapperStyle: ViewPropTypes.style,
};

const defaultProps = {
  statusBarStyle: 'dark-content',
  statusBarColor: styleConstants.colors.white,
};

const HeaderBar = ({
  showShadow,
  statusBarStyle,
  statusBarColor,
  leftComponent,
  leftIconName,
  handleLeftIconPress,
  textComponent,
  text,
  textLeft,
  textRight,
  handleTextPress,
  rightComponent,
  rightIconName,
  handleRightIconPress,
  androidRipple,
  androidRippleColor,
  androidRippleBorderless,
  leftIconStyle,
  textStyle,
  rightIconStyle,
  style,
  wrapperStyle,
}) => {
  const iOSStyles = Platform.OS === 'ios' && { marginTop: 22 }; // iOSStatusBar
  const showShadowStyles = showShadow && styleConstants.shadows.regular;
  const textLeftStyles = textLeft && {
    alignItems: 'flex-start',
    paddingLeft: styleConstants.dimensions.padding.small,
  };
  const textRightStyles = textRight && { alignItems: 'flex-end' };

  let headerLeftComponent;
  let headerTextComponent;
  let headerRightComponent;

  if (leftComponent) {
    headerLeftComponent = leftComponent;
  } else if (leftIconName) {
    headerLeftComponent = (
      <Touchable
        style={textLeft ? { justifyContent: 'center' } : styles.leftIconContainer}
        onPress={handleLeftIconPress}
        disabled={!handleLeftIconPress}
        androidRipple={androidRipple}
        androidRippleColor={androidRippleColor}
        androidRippleBorderless={androidRippleBorderless}
      >
        <MaterialIcon name={leftIconName} style={[styles.leftIcon, leftIconStyle]} />
      </Touchable>
    );
  } else if (!textLeft) {
    headerLeftComponent = <View style={styles.leftIconContainer} />;
  }

  if (textComponent) {
    headerTextComponent = textComponent;
  } else if (text) {
    headerTextComponent = (
      <Touchable
        style={[styles.textContainer, textLeftStyles, textRightStyles]}
        onPress={handleTextPress}
        disabled={!handleTextPress}
        androidRipple={androidRipple}
        androidRippleColor={androidRippleColor}
        androidRippleBorderless={androidRippleBorderless}
      >
        <Text style={[styles.text, textStyle]} numberOfLines={1}>
          {text}
        </Text>
      </Touchable>
    );
  }

  if (rightComponent) {
    headerRightComponent = rightComponent;
  } else if (rightIconName) {
    headerRightComponent = (
      <Touchable
        style={textRight ? { justifyContent: 'center' } : styles.rightIconContainer}
        onPress={handleRightIconPress}
        disabled={!handleRightIconPress}
        androidRipple={androidRipple}
        androidRippleColor={androidRippleColor}
        androidRippleBorderless={androidRippleBorderless}
      >
        <MaterialIcon name={rightIconName} style={[styles.rightIcon, rightIconStyle]} />
      </Touchable>
    );
  } else if (!textRight) {
    headerRightComponent = <View style={styles.rightIconContainer} />;
  }

  return (
    <View style={[styles.wrapper, wrapperStyle]}>
      <StatusBar backgroundColor={statusBarColor} barStyle={statusBarStyle} />
      <View style={[styles.container, showShadowStyles, iOSStyles, style]}>
        {headerLeftComponent}
        {headerTextComponent}
        {headerRightComponent}
      </View>
    </View>
  );
};

HeaderBar.propTypes = propTypes;
HeaderBar.defaultProps = defaultProps;

export default HeaderBar;
