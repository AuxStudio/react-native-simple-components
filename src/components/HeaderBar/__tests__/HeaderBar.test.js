/* eslint-disable import/first */
import React from 'react';
import renderer from 'react-test-renderer';
import MaterialIcon from 'react-native-vector-icons/MaterialIcons';

import HeaderBar from '../';

it('renders a HeaderBar', () => {
  expect(
    renderer.create(
      <HeaderBar
        showShadow
        statusBarColor="red"
        statusBarStyle="dark-content"
        leftIconName="check"
        handleLeftIconPress={jest.fn()}
        text="Test"
        rightIconName="close"
        handleRightIconPress={jest.fn()}
        androidRipple
        androidRippleColor="red"
        androidRippleBorderless
        leftIconStyle={{ color: 'red' }}
        textStyle={{ color: 'red' }}
        rightIconStyle={{ color: 'red' }}
        style={{ backgroundColor: 'blue' }}
        wrapperStyle={{ backgroundColor: 'green' }}
      />,
    ),
  ).toMatchSnapshot();
});

it('renders a HeaderBar with a custom left component', () => {
  expect(
    renderer.create(<HeaderBar leftComponent={<MaterialIcon name="check" />} />),
  ).toMatchSnapshot();
});
it('renders a HeaderBar with a custom text component', () => {
  expect(
    renderer.create(<HeaderBar textComponent={<MaterialIcon name="check" />} />),
  ).toMatchSnapshot();
});
it('renders a HeaderBar with a custom right component', () => {
  expect(
    renderer.create(<HeaderBar rightComponent={<MaterialIcon name="check" />} />),
  ).toMatchSnapshot();
});
it('renders a HeaderBar with a text left', () => {
  expect(renderer.create(<HeaderBar text="Test" textLeft />)).toMatchSnapshot();
});
it('renders a HeaderBar with a text left', () => {
  expect(renderer.create(<HeaderBar text="Test" textRight />)).toMatchSnapshot();
});

it('renders a HeaderBar with no props', () => {
  expect(renderer.create(<HeaderBar />)).toMatchSnapshot();
});
/* eslint-enable */
