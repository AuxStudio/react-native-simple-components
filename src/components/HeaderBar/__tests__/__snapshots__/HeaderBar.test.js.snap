// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a HeaderBar 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "stretch",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 100,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 22,
          },
          Object {
            "backgroundColor": "red",
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "stretch",
          "backgroundColor": "#3D84A8",
          "flexDirection": "row",
          "height": 56,
          "justifyContent": "space-between",
          "paddingHorizontal": 16,
        },
        Object {
          "borderColor": null,
          "borderWidth": 0,
          "elevation": 6,
          "shadowColor": "#212121",
          "shadowOffset": Object {
            "height": 1,
            "width": 0,
          },
          "shadowOpacity": 0.33,
          "shadowRadius": 2,
        },
        Object {
          "marginTop": 22,
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "color": "#212121",
                "fontSize": 24,
              },
              undefined,
            ],
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "flex": 5,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#212121",
              "fontSize": 17,
              "textAlign": "center",
            },
            undefined,
          ]
        }
      >
        Test
      </Text>
    </View>
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={false}
        ellipsizeMode="tail"
        style={
          Array [
            Object {
              "color": undefined,
              "fontSize": 12,
            },
            Array [
              Object {
                "color": "#212121",
                "fontSize": 24,
              },
              undefined,
            ],
            Object {
              "fontFamily": "Material Icons",
              "fontStyle": "normal",
              "fontWeight": "normal",
            },
          ]
        }
      >
        
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders a HeaderBar with a custom left component 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "stretch",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 100,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 22,
          },
          Object {
            "backgroundColor": "#FFFFFF",
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "stretch",
          "backgroundColor": "#3D84A8",
          "flexDirection": "row",
          "height": 56,
          "justifyContent": "space-between",
          "paddingHorizontal": 16,
        },
        undefined,
        Object {
          "marginTop": 22,
        },
        undefined,
      ]
    }
  >
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          undefined,
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    />
  </View>
</View>
`;

exports[`renders a HeaderBar with a custom right component 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "stretch",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 100,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 22,
          },
          Object {
            "backgroundColor": "#FFFFFF",
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "stretch",
          "backgroundColor": "#3D84A8",
          "flexDirection": "row",
          "height": 56,
          "justifyContent": "space-between",
          "paddingHorizontal": 16,
        },
        undefined,
        Object {
          "marginTop": 22,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          undefined,
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
  </View>
</View>
`;

exports[`renders a HeaderBar with a custom text component 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "stretch",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 100,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 22,
          },
          Object {
            "backgroundColor": "#FFFFFF",
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "stretch",
          "backgroundColor": "#3D84A8",
          "flexDirection": "row",
          "height": 56,
          "justifyContent": "space-between",
          "paddingHorizontal": 16,
        },
        undefined,
        Object {
          "marginTop": 22,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    />
    <Text
      accessible={true}
      allowFontScaling={false}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": undefined,
            "fontSize": 12,
          },
          undefined,
          Object {
            "fontFamily": "Material Icons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      
    </Text>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    />
  </View>
</View>
`;

exports[`renders a HeaderBar with a text left 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "stretch",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 100,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 22,
          },
          Object {
            "backgroundColor": "#FFFFFF",
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "stretch",
          "backgroundColor": "#3D84A8",
          "flexDirection": "row",
          "height": 56,
          "justifyContent": "space-between",
          "paddingHorizontal": 16,
        },
        undefined,
        Object {
          "marginTop": 22,
        },
        undefined,
      ]
    }
  >
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 5,
          "justifyContent": "center",
          "opacity": 1,
          "paddingLeft": 8,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#212121",
              "fontSize": 17,
              "textAlign": "center",
            },
            undefined,
          ]
        }
      >
        Test
      </Text>
    </View>
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    />
  </View>
</View>
`;

exports[`renders a HeaderBar with a text left 2`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "stretch",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 100,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 22,
          },
          Object {
            "backgroundColor": "#FFFFFF",
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "stretch",
          "backgroundColor": "#3D84A8",
          "flexDirection": "row",
          "height": 56,
          "justifyContent": "space-between",
          "paddingHorizontal": 16,
        },
        undefined,
        Object {
          "marginTop": 22,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    />
    <View
      accessible={true}
      isTVSelectable={true}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 5,
          "justifyContent": "center",
          "opacity": 1,
        }
      }
    >
      <Text
        accessible={true}
        allowFontScaling={true}
        ellipsizeMode="tail"
        numberOfLines={1}
        style={
          Array [
            Object {
              "color": "#212121",
              "fontSize": 17,
              "textAlign": "center",
            },
            undefined,
          ]
        }
      >
        Test
      </Text>
    </View>
  </View>
</View>
`;

exports[`renders a HeaderBar with no props 1`] = `
<View
  style={
    Array [
      Object {
        "alignSelf": "stretch",
      },
      undefined,
    ]
  }
>
  <View
    style={
      Array [
        Object {
          "left": 0,
          "position": "absolute",
          "right": 0,
          "top": 0,
          "zIndex": 100,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "height": 22,
          },
          Object {
            "backgroundColor": "#FFFFFF",
          },
        ]
      }
    />
  </View>
  <View
    style={
      Array [
        Object {
          "alignItems": "stretch",
          "backgroundColor": "#3D84A8",
          "flexDirection": "row",
          "height": 56,
          "justifyContent": "space-between",
          "paddingHorizontal": 16,
        },
        undefined,
        Object {
          "marginTop": 22,
        },
        undefined,
      ]
    }
  >
    <View
      style={
        Object {
          "alignItems": "flex-start",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "flex-end",
          "flex": 1,
          "justifyContent": "center",
        }
      }
    />
  </View>
</View>
`;
